### PR TITLE
feat(clerk-expo): Add expo data to request headers

### PR DESCRIPTION
**Pull request by thiskevinwang**
## Description

**What**
Send along `x-expo-execution-environment` and `x-expo-native-application-version` headers to help identify the state of a mobile app, calling Clerk.

**Why**
We want to be able to easily determine if a mobile app is in any of "local development", "test flight", "app store/google play" (aka: dev/staging/production, respectively)

## Checklist

- [ ] `npm test` runs as expected.
- [ ] `npm run build` runs as expected.
- [ ] (If applicable) [JSDoc comments](https://jsdoc.app/about-getting-started.html) have been added or updated for any package exports
- [ ] (If applicable) [Documentation](https://github.com/clerk/clerk-docs) has been updated

## Type of change

- [ ] 🐛 Bug fix
- [ ] 🌟 New feature
- [ ] 🔨 Breaking change
- [ ] 📖 Refactoring / dependency upgrade / documentation
- [ ] other:

## Packages affected

- [ ] `@clerk/backend`
- [ ] `@clerk/chrome-extension`
- [ ] `@clerk/clerk-js`
- [x] `@clerk/clerk-expo`
- [ ] `@clerk/fastify`
- [ ] `gatsby-plugin-clerk`
- [ ] `@clerk/localizations`
- [ ] `@clerk/nextjs`
- [ ] `@clerk/clerk-react`
- [ ] `@clerk/remix`
- [ ] `@clerk/clerk-sdk-node`
- [ ] `@clerk/shared`
- [ ] `@clerk/themes`
- [ ] `@clerk/types`
- [ ] `build/tooling/chore`

### CODE REVIEW

File: packages/expo/src/singleton.ts [hunk 1]
```diff
@@ -41,6 +49,24 @@ export function buildClerk({ key, tokenCache }: BuildClerkOptions): HeadlessBrow
...
+      // [isDevice]: true if the app is running on a real device and false if running in a simulator or emulator. On web, this is always set to true.
+      (requestInit.headers as Headers).set('x-expo-device-is-device', Device.isDevice ? 'true' : 'false');
+      (requestInit.headers as Headers).set('x-expo-device-brand', Device.brand ?? NULL_VALUE);
+      (requestInit.headers as Headers).set('x-expo-device-model-id', Device.modelId ?? NULL_VALUE);
+      (requestInit.headers as Headers).set('x-expo-application-id', Application.applicationId ?? NULL_VALUE);
+      (requestInit.headers as Headers).set('x-expo-application-name', Application.applicationName ?? NULL_VALUE);
+      (requestInit.headers as Headers).set(
+        'x-expo-native-application-version',
+        Application.nativeApplicationVersion ?? NULL_VALUE,
+      );
```
Comment (BRKalow) [inline packages/expo/src/singleton.ts:65]: What insight might we gain from the app name / id?

Comment (thiskevinwang) [reply packages/expo/src/singleton.ts:65]: 💭 app name == `Expo Go` is roughly equivalent to request.origin == `http://localhost:3000`

Comment (thiskevinwang) [reply packages/expo/src/singleton.ts:65]: More generally, my thinking here falls mostly under "I don't know what I don't know" regarding insights. But presence of this data should at least enable gaining any insight at all!

**Review by jescalan — COMMENTED** [review]
This looks good to me, but I'm not sure I understand the context behind why it's being done. The PR description is just a screenshot of the result, the name of the branch honestly is the closest to context that we have, but even checking the related linear issue pulled from the branch name, there isn't really a lot of context about why this is important or where it came from.

To be clear, I trust that it's for a good reason, just some feedback on how to make PRs like this easier to understand and review for folks down the line 😁

**Comment by thiskevinwang** [comment]
@BRKalow I reduced the number of headers sent to 2.

@jescalan I updated the description.

File: .changeset/sixty-rats-bathe.md [hunk 2]
```diff
@@ -0,0 +1,5 @@
+---
+'@clerk/clerk-expo': minor
+---
+
+Include new headers in requests; add peer deps
```
Comment (thiskevinwang) [inline .changeset/sixty-rats-bathe.md:5]:
```suggestion
Include new headers in requests;

Added `expo-application >=5` & `expo-constants >=14` peer deps
```

**Review by BRKalow — APPROVED** [review]
👍

**Comment by auropsoft** [comment]
This change causes issues with the use of clerk/clerk-expo with expo start --web.
The preflight requests get blocked by clerk (standard message in the browser:
`...blocked by CORS policy: Request header field x-expo-native-application-version is not allowed by Access-Control-Allow-Headers in preflight response`
```
Access to fetch at 'https://gorgeous-swift-17.clerk.accounts.dev/v1/environment?_clerk_js_version=4.69.1&_method=PATCH&_is_native=1&__dev_session=eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJkZXYiOiJkdmJfMmNwRFhxOVpWNFpZQzhLVk5lSXBSeklmNHhWIn0.awSF7yCL7AB4HTf1PjBpzEi-uIINoVX8n6u6pkBq2lIQ91MXHFdISlEXLX_G28DOGAYqZReu5NJdKn-MejXBXAdy99ohZ4hxhAMwz5bnRUtnZ94OAFEY2jBBHfsTW9YjNipWjOlTbE1FHJ5QM06hS3TiJaVUx9ZqBOHRGcGIj9_lbHdEfSjrX2lULEiV59ggqWEPl7FOlbDGLdyhqv3CnwxmH65EY-TT2G-12Fues1KA3i2zyOppNONC7cNz1eVuFEnQfavIeg31-vgCqN-SjbEG2-QesQvozE_ABMZxvy_ij4-SP7uRjUCYwLoAu5Z05MMY52ULroVojTbI1CnZkA' from origin 'http://localhost:8081' has been blocked by CORS policy: Request header field x-expo-native-application-version is not allowed by Access-Control-Allow-Headers in preflight response.
```

Adding a bit of code to remove those headers for web platform fixes the issue:
```
    useEffect(() => {
        if(Platform.OS === 'web') {
            clerk.__unstable__onBeforeRequest(async (requestInit: RequestInit) => {
                (requestInit.headers as Headers).delete("x-expo-execution-environment");
                (requestInit.headers as Headers).delete("x-expo-native-application-version");
            },[]);
        }
    })
```
I assume that web, being the underused platform for RN, wasn't specifically tested, but i'd love to be able to run it with the same code as the other platforms.

Does this have to be a seperate report. This feels like a server-side issue, not so much a package issue. One would assume the statistics want to be gathered either way.

**Comment by thiskevinwang** [comment]
Hey @auropsoft, 

> Does this have to be a seperate report?
This feels like a server-side issue, not so much a package issue. One would assume the statistics would want to be gathered either way from the description.

Currently, the `@clerk/expo` SDK only supports mobile app output, and not web, so this flow was not tested. This is not necessarily a bug right now, but a helpful breadcrumb for us nonetheless (thanks for using the Expo SDK 😄 ).

Does this sound reasonable?

---

> [!TIP] 
> You can upvote the addition of Clerk supporting web output [here](https://feedback.clerk.com/roadmap/fb56a8db-763a-4e0d-a4b2-f8a453be195c)!

**Pull request merged**